src/OffloadGPULoops.cpp

```diff
@@ -302,25 +284,26 @@ class InjectGpuOffload : public IRMutator {
 
             // If the module state for this API/function did not get created, there were
             // no kernels using this API.
-            Expr state_ptr = module_state_ptr(api_unique_name, false);
-            if (!state_ptr.defined()) {
+            if (!state_needed[api_unique_name]) {
                 continue;
             }
+            Expr state_ptr = make_state_var(api_unique_name);
+            Expr state_ptr_var = Variable::make(type_of<void *>(), api_unique_name);
 
             debug(2) << "Generating init_kernels for " << api_unique_name << "\n";
             vector<char> kernel_src = i.second->compile_to_src();
-            Expr kernel_src_buf = make_buffer_ptr(kernel_src, api_unique_name + "_kernels");
+            Expr kernel_src_buf = make_buffer_ptr(kernel_src, api_unique_name + "gpu_source_kernels");
```

Comment:
Should there be an underscore here? This will emit something like "openclgpu_source_kernels" I think.